### PR TITLE
Kubernetes Container Resources Management

### LINKED ISSUE #12398: Ability to declare Kubernetes resources requests and limits through properties
**Description**
Kubernetes allows to specify how much resources (memory and cpu) a container needs to be executed. In addition to this, Kubernetes also allows you to enforce a limit on the amount of resources (again memory and cpu) that a container can use.

This information can be provided inside the following structure in the following yaml excerpt:
```
    spec:
      containers:
        image: quay.io/myrepo/myapp:1.0.0-SNAPSHOT
        imagePullPolicy: Always
        resources:
          requests:
            memory: "64Mi"
            cpu: "250m"
          limits:
            memory: "512Mi"
            cpu: "1000m"
```

It'd be nice if the `quarkus-kubernetes` extension could generate these information using some properties. A proposed name and usage for these properties is:
```
quarkus.kubernetes.resources.requests.memory=64Mi
quarkus.kubernetes.resources.requests.cpu=250m
quarkus.kubernetes.resources.limits.memory=512Mi
quarkus.kubernetes.resources.limits.cpu=1000m
```

### PR DESCRIPTION
**Pull request by Vinche59**
Allow cpu & memory limits and requests to be generated in kubernetes deployment

Fix: #12398

Signed-off-by: Vincent Sourtin <sourin-v@bridgestone-bae.com>

### CODE REVIEW

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyMemoryDecorator.java
Comment (iocanel) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyMemoryDecorator.java:10]: Why don't we use the upstream decorators?

Comment (Vinche59) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyMemoryDecorator.java:10]: Hello @iocanel 

Because as for now, the generated resources for memory limits/requests is named "**mem**" and not "**memory**" in dekorate.
This was the goal of my PR: https://github.com/dekorateio/dekorate/pull/653 you merged into dekorate some days ago.

Until this fix is incorporated into a dekorate release and the new version bumped into Quarkus, we have to use that custom decorator.
Once done, of course, I will correct to use the upstream decorator.

**Comment by Vinche59** [comment]
@iocanel @geoand 
Can you review and tell me if it's ok to merge as-is or if I have to change something ?

Thanks.

**Comment by geoand** [comment]
I'll leave this to @iocanel as he did the initial review

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyMemoryDecorator.java
Comment (Vinche59) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyMemoryDecorator.java]: Hello @iocanel 

Now that upstream fix have been integrated into Quarkus, I remove the custom decorator for "requests memory".

**Review by iocanel — APPROVED** [review]
LGTM

**Comment by iocanel** [comment]
@Vinche59: Thanks for all the work you put on this!

**Comment by yanaga** [comment]
Kudos to @Vinche59! 👏👏👏

**Comment by Vinche59** [comment]
Don't thank me, I was also in need of this functionality :-)
By the way, thanks all of you for this great piece of software.

**Pull request merged**
